src/Chapter2.hs

```diff
@@ -857,7 +913,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (x:xs) = rewind xs ++ [x]
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Thank you! 😄  for the suggestions. I read the post and this is what I came up with...Is this correct?
``` Haskell
rewindGo :: [a] -> [a]
rewindGo = go []
    where
        go :: [a] -> [a] -> [a]
        go acc [] = acc
        go acc (x:xs) = go (x:acc) xs
```
After reading the above-linked section on the recursive-go, I have a few questions. Some of those are somewhat resolved while working through Chapter 3 while others aren't. Can I ask those in the PR for Chapter3 solutions?

Reply:
Yes, this is correct 👏🏼 

Of course! You can ask them in Chapter 3 as well, no worries 👍🏼